test/RibbonDeltaVault.ts

```diff
@@ -0,0 +1,2782 @@
+
+        await rollToNextOption();
+
+        await vault.initiateWithdraw(depositAmount);
+      });
+
+      it("reverts when not initiated", async function () {
+        await expect(
+          vault.connect(ownerSigner).completeWithdraw()
+        ).to.be.revertedWith("Not initiated");
+      });
+
+      it("reverts when round not closed", async function () {
+        await expect(vault.completeWithdraw()).to.be.revertedWith(
+          "Round not closed"
+        );
+      });
+
+      it("reverts when calling completeWithdraw twice", async function () {
+        await rollToSecondOption(firstOptionStrike);
+
+        await vault.completeWithdraw();
+
+        await expect(vault.completeWithdraw()).to.be.revertedWith(
+          "Not initiated"
+        );
+      });
+
+      it("completes the withdrawal", async function () {
+        const firstStrikePrice = firstOptionStrike;
+        const settlePriceITM = isPut
+          ? firstStrikePrice.sub(100000000)
+          : firstStrikePrice.add(100000000);
+
+        await rollToSecondOption(settlePriceITM);
+
+        const pricePerShare = await vault.roundPricePerShare(2);
+        const withdrawAmount = depositAmount
+          .mul(pricePerShare)
+          .div(BigNumber.from(10).pow(await vault.decimals()));
+
+        let beforeBalance: BigNumber;
+        if (collateralAsset === WETH_ADDRESS) {
+          beforeBalance = await provider.getBalance(user);
+        } else {
+          beforeBalance = await assetContract.balanceOf(user);
+        }
+
+        const tx = await vault.completeWithdraw({ gasPrice });
+        const receipt = await tx.wait();
+        const gasFee = receipt.gasUsed.mul(gasPrice);
+
+        await expect(tx)
+          .to.emit(vault, "Withdraw")
+          .withArgs(user, withdrawAmount.toString(), depositAmount);
+
+        if (collateralAsset !== WETH_ADDRESS) {
+          const collateralERC20 = await getContractAt(
+            "IERC20",
+            collateralAsset
+          );
+
+          await expect(tx)
+            .to.emit(collateralERC20, "Transfer")
+            .withArgs(vault.address, user, withdrawAmount);
+        }
+
+        const { initiated, shares, round } = await vault.withdrawals(user);
+        assert.isFalse(initiated);
+        assert.equal(shares, 0);
+        assert.equal(round, 2);
+
+        let actualWithdrawAmount: BigNumber;
+        if (collateralAsset === WETH_ADDRESS) {
+          const afterBalance = await provider.getBalance(user);
+          actualWithdrawAmount = afterBalance.sub(beforeBalance).add(gasFee);
+        } else {
+          const afterBalance = await assetContract.balanceOf(user);
+          actualWithdrawAmount = afterBalance.sub(beforeBalance);
+        }
+        // Should be less because the pps is down
+        assert.bnLt(actualWithdrawAmount, depositAmount);
+        assert.bnEqual(actualWithdrawAmount, withdrawAmount);
+      });
+    });
+
+    describe("#setStrikePrice", () => {
```

Comment:
Is setting the strike price relevant for DeltaVault? Because we're just mirroring what the option is on Theta Vault

Reply:
i copy pasta-d thetavault test suite. youre right most of these tests not relevant. now that you commented on logic removing now